Library/ViewModels/ConfirmDetailsViewModel.swift

```diff
@@ -362,10 +362,14 @@ public class ConfirmDetailsViewModel: ConfirmDetailsViewModelType, ConfirmDetail
             return [String](repeating: reward.graphID, count: count)
           }
 
+        let locationId = selectedShippingRule == nil
```

Comment:
Suggestion: You could do `selectedShippingRule.flatMap { String($0.location.id) }` to make this one line and remove the `!` below (and do the same for `NoShippingConfirmDetailsViewModel`

Reply:
I like that! Thanks!